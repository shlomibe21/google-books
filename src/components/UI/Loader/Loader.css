.spinnerModal {
    position: fixed;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    text-align: center;
    z-index: 1; /* Sit on top */
    overflow: auto;
    background: url("./images/ajax-indicator-circle-ball-blue.gif") no-repeat
      center fixed;
    background-color: rgba(
      255,
      255,
      255,
      0.1
    ); /* Create overlay - black w/ opacity */
  }
  s
  .spinnerModal p {
    margin-top: 300px;
  }