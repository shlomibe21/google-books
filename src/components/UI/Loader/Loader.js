import React from 'react';

const Loader = () => (
  <div id='loader'><i className='spinnerModal'></i></div>
)

export default Loader;