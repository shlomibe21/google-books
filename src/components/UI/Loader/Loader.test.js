import React from "react";
import Loader from "./Loader";
import { shallow } from "enzyme";

describe("<Loader />", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Loader />);
  });

  it("Matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders a single font-awesome spinner icon", () => {
    expect(wrapper.find("i").prop("className")).toEqual("spinnerModal");
  });
});