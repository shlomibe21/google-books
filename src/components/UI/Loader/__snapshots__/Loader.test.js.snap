// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Matches snapshot 1`] = `ShallowWrapper {}`;
